noir/noir-repo/acvm-repo/brillig_vm/src/lib.rs

```diff
@@ -449,17 +449,16 @@ impl<'a, F: AcirField, B: BlackBoxFunctionSolver<F>> VM<'a, F, B> {
                         }
                         HeapValueType::Array { value_types, size } => {
                             let array_address = self.memory.read_ref(value_address);
-                            let array_start = self.memory.read_ref(array_address);
```

Comment:
Curious, what was it before? we read the array start from the array address?

Reply:
So previously the runtime needed to do a double dereference to get to the items. Now it's a single dereference and add one, to reach the items pointer

Reply:
For vectors now it's also very similar:
```
stack: parent ptr
parent ptr: [parent RC, parent size, ptr_a, ptr_b, ...]
ptr_a: [a RC, a size, a_0, a_1, a_2...]
```